log4j-core/src/main/java/org/apache/logging/log4j/core/appender/nosql/NoSqlAppender.java

```diff
@@ -74,9 +74,6 @@ public NoSqlAppender build() {
 
             final NoSqlDatabaseManager<?> manager = NoSqlDatabaseManager.getNoSqlDatabaseManager(managerName,
                     bufferSize, provider);
-            if (manager == null) {
-                return null;
-            }
```

Comment:
It must be there for a reason but we don't have a unit test that fails without it maybe. It might have  to do with the reconnection logic which can be hard to test.

Reply:
IntelliJ's static code analysis noted that `manager` is never `null`.